controllers/updateservice_controller.go

```diff
@@ -516,16 +516,36 @@ func validateRouteName(instance *cv1.UpdateService, name string, namespace strin
 }
 
 func (r *UpdateServiceReconciler) ensurePolicyEngineRoute(ctx context.Context, reqLogger logr.Logger, instance *cv1.UpdateService, resources *kubeResources) error {
+	oldRoute := resources.policyEngineOldRoute
 	route := resources.policyEngineRoute
+
 	// Set UpdateService instance as the owner and controller
 	if err := controllerutil.SetControllerReference(instance, route, r.Scheme); err != nil {
 		return err
 	}
 
-	// Check if it already exists
+	// The route has changed from old name to new name. So we need to check if either routes
+	// exists
 	found := &routev1.Route{}
-	err := r.Client.Get(ctx, types.NamespacedName{Name: route.Name, Namespace: route.Namespace}, found)
+	foundOldRoute := &routev1.Route{}
+	oldRoutePresent := true
+
+	err := r.Client.Get(ctx, types.NamespacedName{Name: oldRoute.Name, Namespace: oldRoute.Namespace}, foundOldRoute)
 	if err != nil && errors.IsNotFound(err) {
+		oldRoutePresent = false
+	} else if err != nil {
+		handleErr(reqLogger, &instance.Status, "GetRouteFailed", err)
+		return err
+	}
+
+	if uri, _, err := routeapihelpers.IngressURI(foundOldRoute, ""); err == nil {
+		instance.Status.PolicyEngineURI = uri.String()
+	} else {
+		handleErr(reqLogger, &instance.Status, "RouteIngressFailed", err)
+	}
+
+	err = r.Client.Get(ctx, types.NamespacedName{Name: route.Name, Namespace: route.Namespace}, found)
```

Comment:
Seems we should check `oldRoutePresent` and if `true` not bother with this.

Reply:
Right. good catch. Will fix it.